App/BraveWidgets/News Topics/NewsTopicsModel.swift

```diff
@@ -0,0 +1,163 @@
+// Copyright 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import CodableHelpers
+import Collections
+import UIKit
+
+/// Handles fetching Brave News Topics for widgets
+struct NewsTopicsModel {
+  var fetchNewsTopics: @Sendable () async -> [NewsTopic]
+  var fetchImageThumbnailsForTopics: @Sendable ([NewsTopic], CGSize) async -> [NewsTopic.ID: UIImage]
+}
+
+extension NewsTopicsModel {
+  /// The live implementation of the news topic model
+  ///
+  /// - note: For now this implementation does not cache the results of the file, since the widget is not
+  /// reloaded by the app in any way yet. Hopefully we can move this code to the BraveNews module in the
+  /// future without bloating the size of the widget extension and then get all the cacheing mechanisms
+  /// from there.
+  static var live: Self {
+    .init(
+      fetchNewsTopics: {
+        do {
+          // At the moment there is only english US topics
+          let url = URL(string: "https://brave-today-cdn.brave.com/news-topic-clustering/widget_topic_news.en_US.json")!
```

Comment:
Will this URL always be the same and for en_US only? If it's possible that it won't be ASCII, we should use the `NSURL(idnString:)`

Reply:
This URL will always be this structure, the only thing that may change in the future is that the `en_US` locale + region may switch to language/region specifics for the user. No worries that it won't ever be ascii!